ui/perfherder/alerts/CollapsableRows.jsx

```diff
@@ -0,0 +1,132 @@
+  topRows = 20;
+
+  bottomRows = 6;
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      isOpen: false,
+    };
+  }
+
+  toggleRows = () => {
+    this.setState((prevState) => ({
+      isOpen: !prevState.isOpen,
+    }));
+  };
+
+  render() {
+    const { filteredAndSortedAlerts } = this.props;
+    const { isOpen } = this.state;
+
+    const startRows = filteredAndSortedAlerts.slice(0, this.topRows);
+    const middleRows = filteredAndSortedAlerts.slice(
+      this.topRows,
+      filteredAndSortedAlerts.length - this.bottomRows,
+    );
+    const endRows = filteredAndSortedAlerts.slice(
+      filteredAndSortedAlerts.length - this.bottomRows,
+      filteredAndSortedAlerts.length,
+    );
+
+    return (
+      <React.Fragment>
+        {startRows.map((alert) => (
+          <AlertTableRow key={alert.id} alert={alert} {...this.props} />
+        ))}
+        {isOpen && (
+          <React.Fragment>
+            <tr
+              className="border-left border-right"
+              role="button"
+              onClick={() => this.toggleRows()}
+              data-testid="show-less-alerts"
+            >
+              <td />
+              <td />
+              <td>
+                <span className="cursor-pointer">
+                  <FontAwesomeIcon icon={faAngleUp} /> Show less alerts
+                </span>
+              </td>
+              <td />
+              <td />
+              <td />
+              <td />
+            </tr>
+            {middleRows.map((alert) => (
+              <AlertTableRow key={alert.id} alert={alert} {...this.props} />
+            ))}
+          </React.Fragment>
+        )}
+        {!isOpen && (
+          <React.Fragment>
+            <tr
+              className="border-left border-right cursor-pointer"
+              role="button"
+              onClick={() => this.toggleRows()}
+              data-testid="show-more-alerts"
+            >
+              <td className="cursor-pointer" />
+              <td />
+              <td>
+                <span className="cursor-pointer">
+                  <SimpleTooltip text="..." tooltipText="Show more alerts" />
+                </span>
+              </td>
+              <td>
+                <span className="cursor-pointer">
+                  <SimpleTooltip text="..." tooltipText="Show more alerts" />
+                </span>
+              </td>
+              <td>
+                <span className="cursor-pointer">
+                  <SimpleTooltip text="..." tooltipText="Show more alerts" />
+                </span>
+              </td>
+              <td>
+                <span className="cursor-pointer">
+                  <SimpleTooltip text="..." tooltipText="Show more alerts" />
+                </span>
+              </td>
+              <td>
+                <span className="cursor-pointer">
+                  <SimpleTooltip text="..." tooltipText="Show more alerts" />
+                </span>
```

Comment:
Instead of using the "..." to show more alerts, maybe we can use something similar to "Show less alerts", check the attached screenshot.

In this way you get rid of the repetitive code and it is more intuitive.

![image](https://user-images.githubusercontent.com/88494730/130613877-d54a3b33-fac2-4b42-8dba-b8028ba1f738.png)

Reply:
This sounds like a good idea. I agree, this looks more intuitive to users.